woocommerce/woocommerce
# use page.waitForTimeout and page.waitForXPath instead of deprecated page.waitFor

In the e2e-utils package, there are a few instances of `page.waitFor`. This [has been deprecated](https://pptr.dev/#?product=Puppeteer&version=v11.0.0&show=api-pagewaitforselectororfunctionortimeout-options-args) and is generating a lot of noise in the tests. This PR upgrades any uses of `page.waitFor()` to more specific functions like `page.waitForTimeout()` and `page.waitForXPath()`. I also removed the custom `waitForTimeout` method and replaced it with is only used in the puppeteer context and is the same as `page.waitForTimeout()`

### All Submissions:






### Changes proposed in this Pull Request:



### How to test the changes in this Pull Request:

The only way I can think of testing this is to `npm link` this package locally into another repository that depends on the e2e-utils package. Since I work on WooCoommerce Blocks, I'll suggest using this one, but this can be done with any other repo that relies heavily on the `e2e-test` package. Granted it's not a fool proof test, but given this is quite a trivial change, I don't think it's worth creating custom tests to check every util function where `waitFor` is used.

1. Run the WooCommerce plugin locally
2. Clone the [WooCommerce Blocks](https://github.com/woocommerce/woocommerce-blocks) plugin locally
3. Open a terminal window and navigate to the `e2e-utils` package directory. 
4. Run `pnpm install && npm link` 
6. Open another terminal window, navigate to the WooCommerce Blocks repository
7. `composer install && npm i && npm run build && npm link @woocommerce/e2e-utils`
8.  `npm run test:e2e -- checkout.test.js`
9. Some tests might fail currently but what is important to check here is that there are not console warnings equivalent to the following:
```
waitFor is deprecated and will be removed in a future release. See https://github.com/puppeteer/puppeteer/issues/6214 for details and how to migrate your code.
```

### Other information:




### FOR PR REVIEWER ONLY:
